scipp-imaging-2D.ipynb

```diff
@@ -224,218 +162,516 @@
    },
    "outputs": [],
    "source": [
-    "# Load the data into the dataset\n",
-    "ds[\"sample\"] = tiffs_to_variable(sample_path)\n",
-    "ds[\"reference\"] = tiffs_to_variable(ob_path)"
+    "# Adds a coordinate for the spectra and TOF\n",
+    "ds.coords[\"spectrum\"] = sc.Variable([\"spectrum\"], values=np.arange(ds[\"sample\"].shape[1]))\n",
+    "stitched = sc.Dataset(coords={\"tof\": sc.Variable([\"tof\"], unit=sc.units.us,\n",
+    "                              values=np.arange(start=rebin_parameters[\"start\"], stop=rebin_parameters[\"stop\"],\n",
+    "                                               step=rebin_parameters[\"width\"], dtype=np.float64))})"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {
-    "scrolled": false
+    "pycharm": {
+     "name": "#%%\n"
+    }
    },
    "outputs": [],
    "source": [
-    "sc.plot.plot(sc.sum(ds, Dim.Spectrum)) if do_plots else None"
+    "# Pre-calculate frame shifts. This will be replaced with automatic WFM stitching in the future\n",
+    "frame_shifts = [sum(frame_shift_increments[:i + 1]) for i in range(len(frame_shift_increments))]\n",
+    "\n",
+    "def stitch_data(variable_to_stitch):\n",
+    "    print(f\"Stitching: {variable_to_stitch}\")\n",
+    "    stitched[variable_to_stitch] = imaging.stitch(ds[variable_to_stitch],\n",
+    "                                                  frame_parameters=frame_parameters,\n",
+    "                                                  frame_shifts=frame_shifts,\n",
+    "                                                  rebin_parameters=rebin_parameters)\n",
+    "\n",
+    "for var_name in ds.keys():\n",
+    "    stitch_data(var_name)"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {
-    "scrolled": true
+    "pycharm": {
+     "name": "#%%\n"
+    }
    },
    "outputs": [],
    "source": [
-    "# Adds a coordinate for the spectra\n",
-    "ds.coords[Dim.Spectrum] = sc.Variable([Dim.Spectrum], values=np.arange(ds[\"sample\"].shape[1]))\n",
-    "ds"
+    "integrated = sc.Dataset()\n",
+    "for dim in stitched.coords.keys():\n",
+    "    integrated.coords[dim] = stitched.coords[dim]\n",
+    "\n",
+    "integrated = sc.sum(stitched, 'tof')\n",
+    "# Put TOF coords back for the export to TIFF stack later\n",
+    "integrated.coords[\"tof\"] = stitched.coords[\"tof\"]"
```

Comment:
A couple of things:
- First 4 lines redundant, since you just "overwrite" `integrated` with the sum.
- Last line is a bit weird to have as a coord. Have you considered other options?

Reply:
- Good spot, from when I was originally integrating in a loop without using `sum`
- I've moved the coords copying down to the cell where we export tiffs, we TOF values which are written into a .CSV so other tools (like labview) can open the stack correctly. Copying them through seemed easiest and we don't use `integrated` after we have calculate mask positions

Reply:
> Copying them through seemed easiest

Coordinates in scipp are a very particular concept: They are used for ensuring correct and consistent operation. Setting random things are coords is conceptually not meaningful and will likely break some things later on. Can't you stoe it separately?